lib/worker.js

```diff
@@ -0,0 +1,61 @@
+var debug       = require('debug')('github:worker');
+var tcrc        = require('./taskclusterrc');
+var github      = require('./github');
+var request     = require('request-promise')
+var taskcluster = require('taskcluster-client')
+var slugid      = require('slugid');
+
+var worker = module.exports = {};
+
+/**
+ * If a .taskclusterrc exists, attempt to turn it into a taskcluster
+ * graph config, and submit it to the scheduler.
+ **/
+worker.webHookHandler = async function(message, context) {
+  debug('handling pull request: ', message);
+  let msgPayload = message.payload;
+  // Try to fetch a taskclusterrc file for every request
+  let taskclusterrc = await request(tcrc.buildConfigUrl(msgPayload));
+  if (taskclusterrc) {
+    try {
+      let scheduler = new taskcluster.Scheduler(context.cfg.get('taskcluster'));
+      let graphConfig = await tcrc.processConfig(taskclusterrc, msgPayload);
+      let graphStatus = await scheduler.createTaskGraph(slugid.v4(), graphConfig);
+    } catch(e) {
+      debug(e);
+      // Let the user know that there was a problem processing their
+      // config file
+      let statusUpdate = await github.updateStatus(context.githubAPI, {
+          user:        msgPayload.organization,
+          repo:        msgPayload.repository,
+          sha:         msgPayload.details.headSha,
+          state:       'error',
+          description: e.toString()});
+    }
+  };
+};
+
+/**
+ * Post updates to GitHub, when the status of a task changes.
+ **/
+worker.graphStateChangeHandler = async function(message, context) {
```

Comment:
Love this.  Lot cleaner and more informative than just the plain comments that get added in other projects.